app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt

```diff
@@ -531,6 +545,36 @@ class AttendeeFragment : Fragment() {
                 })
     }
 
+    private fun startPayPal() {
+        val payPalConfiguration = PayPalConfiguration()
+            .environment(PayPalConfiguration.ENVIRONMENT_PRODUCTION) //Set to LIVE API CREDENTIALS
+            .clientId(PAYPAL_API_KEY) //LIVE API Client ID
+        Toast.makeText(context, amount.toString(), Toast.LENGTH_SHORT).show()
+        val payment = PayPalPayment(BigDecimal(amount.toString()), paymentCurrency, "Pay for tickets",
+            PayPalPayment.PAYMENT_INTENT_SALE)
+        val intent = Intent(context, PaymentActivity::class.java)
+        intent.putExtra(PayPalService.EXTRA_PAYPAL_CONFIGURATION, payPalConfiguration)
+        activity?.startService(intent)
+        intent.putExtra(PaymentActivity.EXTRA_PAYMENT, payment)
+        startActivityForResult(intent, PAYPAL_REQUEST_CODE)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == PAYPAL_REQUEST_CODE) {
+            if (resultCode == Activity.RESULT_OK) {
+                val paymentConfirmation = data?.getParcelableExtra<PaymentConfirmation>(PaymentActivity.EXTRA_RESULT_CONFIRMATION)
+                if (paymentConfirmation != null) {
+                    val paymentInfo = paymentConfirmation.toJSONObject()
+                    Timber.d(paymentInfo.toString(4))
+                    // TODO: Send token to server and redirect to tickets
```

Comment:
@iamareebjamal I need help here. I'm getting following response - 
```
{
                        "client": {
                            "environment": "sandbox",
                            "paypal_sdk_version": "2.16.0",
                            "platform": "Android",
                            "product_name": "PayPal-Android-SDK"
                        },
                        "response": {
                            "create_time": "2018-12-30T19:59:20Z",
                            "id": "PAY-0JW58086WS3706240LQUSHEQ",
                            "intent": "sale",
                            "state": "approved"
                        },
                        "response_type": "payment"
                    }
```
And according to api documentation for paypal payment request - 
```
RequestHide
Headers
Accept: application/vnd.api+json
Authorization: JWT <Auth Key>
Body
{
  "data": {
    "attributes": {
      "return_url": "https://localhost:5000/return_url",
      "cancel_url": "https://localhost:5000/cancel_url"
    },
    "type": "paypal-payment"
  }
}
```

How to send confirmation token to the server?

Reply:
Are you using Braintree SDK?

Reply:
No, It's PayPal Android SDK.

Reply:
I think Braintree can replace both, Stripe and PayPal.

Reply:
Will see

Reply:
@liveHarshit from what I understand, the server already implements everything it needs, for now, For a Paypal payment, 2 things are needed: A Payer ID and A Payment ID. The Payment ID is generated by the server and I can see that you can get it from PaymentConfirmation above. For Payer ID, please check if you can generate it with the Paypal SDK, probably there should be a Paypal button for the user to log in or some similar method. After you have the Payer ID and the Payment ID, send a Charge object to the server. I think it should work, I will try Paypal SDK for Android to see if I understand it correctly

Reply:
So payer id is the something id of user's PayPal account?

Reply:
Yes, it is the ID of who pay the payment, and it is unchanged while for PayPal user email or relevant information can be changed

Reply:
I think we should Braintree, read comments in this question: https://stackoverflow.com/questions/38229929/getting-the-payerid-from-paypal-android-sdk

Reply:
As it is configured on the server, I don't think we can use it on Android. It is configured to accept approval url.
 https://github.com/paypal/PayPal-node-SDK/issues/24

Reply:
let's me spend a bit more time on this and then we can discuss what solution should we go on this tomorrow meeting, besides, I also think Braintree is better as the old documentation is deleted

Reply:
> I also think Braintree is better as the old documentation is deleted

Exactly, that is the problem. PayPal has deleted the documentation related to that and archive the sdk repo.